app/controllers/api/encryptables_controller.rb

```diff
@@ -165,23 +169,18 @@ def fetch_encryptable_files
   def build_encryptable_file
     filename = params[:file].original_filename
 
-    file = new_file(file_credential, params[:description], filename)
+    file = Encryptable::File.new(description: params[:description], name: filename)
+    file.encryptable_credential = credential if credential_id.present?
+    file.folder = folder if credential_id.blank?
     file.content_type = params[:file].content_type
     file.cleartext_file = params[:file].read
-
     instance_variable_set(:"@#{ivar_name}", file)
   end
 
-  def new_file(parent_encryptable, description, name)
-    Encryptable::File.new(encryptable_credential: parent_encryptable,
-                          description: description,
-                          name: name)
-  end
-
   def credential_id
     return params[:id] if params[:id].present?
 
-    params[:credential_id]
+    params[:credential_id] == 'null' ? nil : params[:credential_id]
```

Comment:
dang does the ember really send `"null"` for a null value? I feel like that shouldn't happen
If it really does, maybe it would make sense to make a method `nil_param?` or something for the ApplicationController and then use it here